### PR TITLE
CBG-1942: Review and update REST API spec

### PR DESCRIPTION
**Pull request by bbrks**
Going to be largely unreviewable due to reordering of properties (via `openapi-format`)

## Changes:
* Run through spellcheck
* Run through openapi-format tool
* Remove implementation details
* Fix /_replicationStatus paths (to include /{db}/)
* Cleanup schema for /
* Improve schema for GET /{db}/
* Clarify non-functionality of /_ensure_full_commit
* Add expvar schema
* Update all pprof endpoints
* Update db config etag/ifmatch descriptions
* Fix incorrect 'value' nesting for single exmaples
* Extract 412 response for db config endpoints
* Fixes to sync, import, resync, purge, etc.

### CODE REVIEW

File: docs/api/api_docs.yaml [hunk 1]
```diff
@@ -24,28 +57,41 @@ paths:
                 type: object
                 properties:
                   ADMIN:
+                    description: '`true` if the request is from the Admin API - otherwise omitted.'
                     type: boolean
-                    description: Whether the request is from the Admin API (`true`) or the Public API (`false`).
+                    example: true
                   couchdb:
+                    description: ''
```
Comment (IsaacLambat) [inline docs/api/api_docs.yaml]: Is it valid to remove empty descriptions from here?

File: docs/api/api_docs.yaml [hunk 2]
```diff
@@ -24,28 +57,41 @@ paths:
...
                     type: string
-                    default: Welcome
+                    example: Welcome
                   vendor:
-                    $ref: '#/components/schemas/Vendor'
+                    description: ''
```
Comment (IsaacLambat) [inline docs/api/api_docs.yaml]: Empty desc

**Review by IsaacLambat — DISMISSED** [review]
LGTM

File: docs/api/api_docs.yaml [hunk 3]
```diff
@@ -731,123 +779,125 @@ paths:
...
+        - Public
   '/{db}/_ensure_full_commit':
     parameters:
       - $ref: '#/components/parameters/db'
     post:
+      summary: ''
+      description: This endpoint is non-functional but is present for CouchDB compatibility.
       responses:
         '201':
           description: ''
```
Comment (bbrks) [inline docs/api/api_docs.yaml]:
```suggestion
          description: 'OK'
```

File: docs/api/api_docs.yaml (hunk 1 above)
Comment (bbrks) [inline docs/api/api_docs.yaml]:
```suggestion
                    description: 'CouchDB welcome'
```

File: docs/api/api_docs.yaml (hunk 2 above)
Comment (bbrks) [inline docs/api/api_docs.yaml]:
```suggestion
                    description: 'Product vendor'
```

**Review by IsaacLambat — APPROVED** [review]

**Pull request merged**